proposals/manual-bundling.md

```diff
@@ -0,0 +1,150 @@
+   
+   In addition to `/platforms`, there's also a `/plugins` directory. This contains the following:
+   
+   - Copies of the plugins that already exist in `node_modules`, but with the dotfiles missing.
+   - `android.json` — This appears to duplicate some info from `package.json` and `config.xml`, but in another format.
+   - `fetch.json` — This appears to duplicate some info from `package.json`, but in yet another format.
+
+
+## Proposal
+
+As a modern JavaScript developer, I would like to use *<this year's popular tool>* to bundle my dependencies and code.
+
+- Get rid of `cordova.js`.
+- Get rid of `www/`.
+- Get rid of `plugins/`.
+
+### 1. Load plugins from the `node_modules` directory, not `plugins`
+
+Cordova already downloads packages to `node_modules` using `cordova-fetch`. There's no need to maintain a copy of them in the `plugins` directory.
+
+This also allows plugin creators to use npm dependencies for both hooks and browser code the way this would be done for other packages.
+
+### 2. Let a developer require Cordova using its own build tools.
+
+Based on [cordova-plugin-camera] as an example:
+
+As a developer, I'd like to turn this:
+
+```js
+navigator.camera.getPicture(cameraSuccess, cameraError, cameraOptions)
+```
+
+Into this:
+
+```js
+import CordovaCamera from 'cordova-plugin-camera';
+
+
+CordovaCamera.getPicture(cameraSuccess, cameraError, cameraOptions);
+```
+
+This can be done by adding the following to `package.json`:
+
+```json
+  "main": "www/Camera.js"
+```
+
+### 3. Use [cordova-js] as a proxy to the platform
+
+Instead of using a custom merge of [cordova-js] and *cordova-`<platform name>`*, the Cordova platform would register itself at runtime, so *cordova-js* basically works as a proxy to the registered platform. For example:
+
+In [cordova-android] `package.json`:
+
+```json
+  "main": "cordova-src-js/index.js"
+```
+
+In [cordova-android] `index.js`:
+
+```js
+const cordova = require('cordova-js');
+
+const platform = require('./platform');
+
+
+cordova.setPlatform(platform);
+```
+
+In the Cordova app code:
+
+```js
+// CORDOVA_PLATFORM is set by the build tool, e.g. using webpack.DefinePlugin.
+require(CORDOVA_PLATFORM)
+```
+
+Plugins would make calls to `cordova-js/exec` (replacing `cordova/exec`) independant of the platform. If a developer wishes to use an outdated plugin, it's the developer's responsibility to alias `cordova/*` calls to `cordova-js/*`.
+
+
+### 4. Dependency management
+
+Cordova doesn't need to define special dependency management for plugins. `npm` can handle this.
+
+Cordova plugins and platforms depend on the same instance of `cordova-js`. For this reason all Cordova platforms and plugins should specify `cordova-js` in their `package.json`'s `peerDependencies`.
+
+Also if plugins depend on other plugins, this can be handled by specifying them as `peerDependencies`.
+
+
+### 5. Fix up the browserify setup
```

Comment:
the browserify code is confusing. With the refactoring of cordova.js we could potentially remove this and allow our devs to use one of the various build step tools in their own setup